run/generate_json_multi_state.py

```diff
@@ -0,0 +1,374 @@
+
+                    # Add to the chain_sequence
+                    chain_seqs[chain.id].append(self.AA3_to_AA1.get(residue.get_resname(), 'XXX'))
+
+                    # Update the chain index
+                    res_index_chain += 1
+
+                # Make the list of AA1s into a single string for the chain sequence
+                chain_seqs[chain.id] = "".join([x for x in chain_seqs[chain.id] if x is not None])
+
+            mutable = []
+            for resind in self.design_res:
+                res_id = resind[0] + str(resind[1])
+                if pdbids[res_id][0] != 'X':
+                    mutable.append({"chain": pdbids[res_id][1], "resid": pdbids[res_id][2], 
+                                    "WTAA": pdbids[res_id][0], "MutTo": self.design_default})
+
+            symmetric = []
+            for symmetry in self.symmetric_res:
+                values = list(symmetry.values())
+
+                for tied_pos in zip(*values):
+                    skip_tie = False
+                    sym_res = []
+                    for pos in tied_pos:
+                        res_id = pos[0] + str(pos[1])
+                        if pdbids[res_id][0] == 'X':
+                            skip_tie = True
+                            break
+                        else:
+                            sym_res.append( pdbids[res_id][1] + str(pdbids[res_id][2]) )
+
+                    if not skip_tie:
+                        symmetric.append(sym_res)
+
+            dictionary = {"sequence" : chain_seqs, "designable": mutable, "symmetric": symmetric, "tied_betas": self.beta_dict, "chain_key": self.chain_dict}
+            pdb_dicts.append(dictionary)
+
+        with open(out_path, 'w') as f:
+            for pdb_dict in pdb_dicts:
+                f.write(json.dumps(pdb_dict, indent=2) + '\n')
+
+
+class NotDisordered(Select):
+    """Checks if atom is disordered, and if so, chooses first entry"""
+    def accept_atom(self, atom):
+        return not atom.is_disordered() or atom.get_altloc() == "A"
+
+
+def get_arguments() -> argparse.Namespace:
+    """ Uses FileArgumentParser to parse commandline options for multi-state design setup."""
+
+    # Construct the parser and its arguments.
+    parser = FileArgumentParser(description='Script that takes a PDB and design'
+                                'specifications to create a json file for input'
+                                'to ProteinMPNN', 
+                                fromfile_prefix_chars='@')
+
+    parser.add_argument('--pdb_dir',
+                        type=str,
+                        help='Path to the directory containing PDB files.')
+    parser.add_argument('--constraints', 
+                        type=str,
+                        default='',
+                        help='comma-separated list of multi-state design constraints. '
+                        'E.g. PDB1:A10-15:1+PDB2:A10-15:0.5+PDB3:B5-10:-3,PDB1:A20-25:1+PDB3:B20-25:1')
```

Comment:
This says a comma-separated list, but from reading the code I think you can have multiple constraints listed here if they're separated by a semicolon (?). If this is true, we should add some documentation about that